src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/KeyController.cs

```diff
@@ -0,0 +1,199 @@
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Catalyst.Core.Modules.Dfs.WebApi.V0.Controllers
+{
+    /// <summary>
+    ///   A cryptographic key.
+    /// </summary>
+    public class CryptoKeyDto
+    {
+        /// <summary>
+        ///   The key's local name.
+        /// </summary>
+        public string Name;
+
+        /// <summary>
+        ///   The key's global unique ID.
+        /// </summary>
+        public string Id;
+    }
+
+    /// <summary>
+    ///   A list of cryptographic keys.
+    /// </summary>
+    public class CryptoKeysDto
+    {
+        /// <summary>
+        ///   A list of cryptographic keys.
+        /// </summary>
+        public IEnumerable<CryptoKeyDto> Keys;
+    }
+
+    /// <summary>
+    ///   A cryptographic key.
+    /// </summary>
+    public class CryptoKeyRenameDto
+    {
+        /// <summary>
+        ///   The key's local name.
+        /// </summary>
+        public string Was;
+
+        /// <summary>
+        ///   The key's global unique ID.
+        /// </summary>
+        public string Now;
+
+        /// <summary>
+        ///   The key's global unique ID.
+        /// </summary>
+        public string Id;
+
+        /// <summary>
+        ///   Indicates that a existing key was overwritten.
+        /// </summary>
+        public bool Overwrite;
+    }
+
+    /// <summary>
+    ///   Manages the cryptographic keys.
+    /// </summary>
+    public class KeyController : IpfsController
+    {
+        /// <summary>
+        ///   Creates a new controller.
+        /// </summary>
+        public KeyController(IDfsService dfs) : base(dfs) { }
+
+        /// <summary>
+        ///   List all the keys.
+        /// </summary>
+        [HttpGet, HttpPost, Route("key/list")]
+        public async Task<CryptoKeysDto> List()
+        {
+            var keys = await IpfsCore.KeyApi.ListAsync(Cancel);
+            return new CryptoKeysDto
+            {
+                Keys = keys.Select(key => new CryptoKeyDto
+                {
+                    Name = key.Name,
+                    Id = key.Id.ToString()
+                })
+            };
+        }
+
+        /// <summary>
+        ///   Create a new key.
+        /// </summary>
+        /// <param name="arg">
+        ///   The name of the key.
+        /// </param>
+        /// <param name="type">
+        ///   "rsa"
+        /// </param>
+        /// <param name="size">
+        ///   The key size in bits, if the type requires it.
+        /// </param>
+        [HttpGet, HttpPost, Route("key/gen")]
+        public async Task<CryptoKeyDto> Create(string arg,
+            string type,
+            int size)
+        {
+            if (String.IsNullOrWhiteSpace(arg))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)